### PR TITLE
hv_crash: function test with hv_crash

### PR DESCRIPTION
**Pull request by kimi1978wy**
QEMU process will quit after nmi with hv_crash flag

id: 1836095

Signed-off-by: Yu Wang <wyu@redhat.com>

### CODE REVIEW

**Comment by kimi1978wy** [comment]
@YongxueHong  Could you help to review this patch? Thanks a lot.

File: qemu/tests/hv_crash.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    flags_with_hv_crash = params["flags_with_hv_crash"]
+    set_nmi_cmd = params["set_nmi_cmd"]
+
+    error_context.context("Boot the guest with hv_crash flag", logging.info)
+    vm = _boot_guest_with_cpu_flag(flags_with_hv_crash)
+
+    error_context.context("Make the guest crash", logging.info)
+    _trigger_crash(vm)
+    logging.info("Check the qemu process is quit")
+    if not utils_misc.wait_for(vm.is_dead, 10, 1, 1):
```
Comment (YongxueHong) [inline qemu/tests/hv_crash.py:57]: `if not utils_misc.wait_for(vm.is_dead, 10, 1, 1):` -> `if not utils_misc.wait_for(vm.is_dead(), 10, 1, 1):`
The vm instance doesn't have the `is_dead` method.

File: qemu/tests/hv_crash.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        test.fail("The qemu still active after crash")
+
+    error_context.context("Boot the guest again", logging.info)
+    vm = _boot_guest_with_cpu_flag(flags_without_hv_crash)
+    vm.verify_alive()
+
+    error_context.context("Make the guest crash again", logging.info)
+    _trigger_crash(vm)
+    logging.info("Check the qemu process is not quit")
```
Comment (YongxueHong) [inline qemu/tests/hv_crash.py:66]: Before here, should we check whether the guest reboots completely? or use `wait_for_login` to check it.
```
  4) use nmi again, the qemu should not quit, the guest will reboot
  5) after reboot, the guest should work well
```

File: qemu/tests/hv_crash.py (hunk 1 above)
Comment (kimi1978wy) [reply qemu/tests/hv_crash.py:56]: I checked the method, it seems that should use utils_misc.wait_for(vm.is_dead, 10, 1, 1).

File: qemu/tests/hv_crash.py (hunk 2 above)
Comment (kimi1978wy) [reply qemu/tests/hv_crash.py:66]: updated, thanks

File: qemu/tests/hv_crash.py (hunk 1 above)
Comment (YongxueHong) [reply qemu/tests/hv_crash.py:56]: Sorry, I mistake it. you are right.

**Comment by kimi1978wy** [comment]
Updated, pls help to review again @YongxueHong  

(1/2) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hv_crash.q35: PASS (364.69 s)
 (2/2) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.hv_crash.q35: PASS (213.30 s)

**Pull request merged**